staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/ratcheting.go

```diff
@@ -408,3 +382,87 @@ func (f ratchetingValueValidator) SetPath(path string) {
 func (f ratchetingValueValidator) Applies(source interface{}, valueKind reflect.Kind) bool {
 	return true
 }
+
+// Key returns the child of the reciever with the given name.
+// Returns nil if the given name is does not exist in the new object, or its
+// value is not correlatable to an old value.
+// If receiver is nil or if the new value is not an object/map, returns nil.
+func (l *CorrelatedObject) Key(field string) *CorrelatedObject {
+	if l == nil || l.schema == nil {
+		return nil
+	} else if existing, exists := l.children[field]; exists {
+		return existing
+	}
+
+	// Find correlated old value
+	oldAsMap, okOld := l.oldValue.(map[string]interface{})
+	newAsMap, okNew := l.value.(map[string]interface{})
+	if !okOld || !okNew {
+		return nil
+	}
+
+	oldValueForField, okOld := oldAsMap[field]
+	newValueForField, okNew := newAsMap[field]
+	if !okOld || !okNew {
+		return nil
+	}
+
+	var propertySchema *spec.Schema
```

Comment:
Will this work against non-structural schemas? Do you expect the schema to be structural here?

Reply:
I'm not sure that matters since we are only traversing the structural nodes of the schema. In later commit this is refactored for `common.Schema` and does indeed work against structural